src/compiler/utilities.ts

```diff
@@ -2239,11 +2239,11 @@ namespace ts {
         return emitOutputFilePathWithoutExtension + extension;
     }
 
-    export function getDeclarationEmitOutputFilePath(sourceFile: SourceFile, host: EmitHost) {
+    export function getDeclarationEmitOutputFilePath(sourceFile: SourceFile, host: EmitHost, ignoreCompilerOptions = false) {

```

Comment:
I think we can make this function always return the right result. Then caller should decide if he needs to call this function is the first place. This function is internal so change in theory should not be observed by anyone